File: python/数组/1700. 无法吃午餐的学生数量.py
from typing import List
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        from collections import deque
        sandwiches = deque(sandwiches)
        students = deque(students)

        while True:
            ok = False
            for _ in range(len(students)):
                if students[0] == sandwiches[0]:
                    students.popleft()
                    sandwiches.popleft()
                    ok = True
                else:
                    students.append(students.popleft())
            if not ok:
                break
        return len(students)

File: go/array/1700. 无法吃午餐的学生数量.go
package array

func countStudents(students []int, sandwiches []int) int {
	for true {
		ok := false
		for i := 0; i < len(students); i++ {
			if students[0] == sandwiches[0] {
				students = students[1:]
				sandwiches = sandwiches[1:]
				ok = true
			} else {
				x := students[0]
				students = students[1:]
				students = append(students, x)
			}
		}
		if !ok {
			break
		}
	}
	return len(students)
}
